dbms/src/Storages/Kafka/StorageKafka.cpp

```diff
@@ -210,8 +210,9 @@ BufferPtr StorageKafka::createBuffer()
     size_t batch_size = max_block_size;
     if (!batch_size)
         batch_size = settings.max_block_size.value;
+    size_t poll_timeout = settings.stream_poll_timeout_ms.totalMilliseconds();
 
-    return std::make_shared<DelimitedReadBuffer>(new ReadBufferFromKafkaConsumer(consumer, log, batch_size), row_delimiter);
+    return std::make_shared<DelimitedReadBuffer>(new ReadBufferFromKafkaConsumer(consumer, log, batch_size, poll_timeout), row_delimiter);
```

Comment:
Looks like possible memory leak. I think it's better to change `DelimitedReadBuffer` constructor (accept unique_ptr there).